pkg/middleware/oapi_validate.go

```diff
@@ -90,7 +99,7 @@ func OapiRequestValidatorWithOptions(swagger *openapi3.T, options *Options) echo
 
 // ValidateRequestFromContext is called from the middleware above and actually does the work
 // of validating a request.
-func ValidateRequestFromContext(ctx echo.Context, router routers.Router, options *Options) error {
+func ValidateRequestFromContext(ctx echo.Context, router routers.Router, options *Options) *echo.HTTPError {
```

Comment:
switched the return type here since it's only called from the above function, all branches return `*echo.HTTPError` types, and returning the concrete type rather than an `error` allows for easier introspection in the `ErrorHandler`.